### PR TITLE
Add pt weights for prompt Ds in Pb-Pb pass3

### PR DESCRIPTION
**Pull request by fgrosa**
- add files with pT weights for Ds in Pb-Pb pass3 (0-10% and 30-50%)
- modify the script such that it reads the configuration from a yaml file

### CODE REVIEW

File: systematics/genptshape/ComputePtGenShapeWeights.py [hunk 1]
```diff
@@ -7,90 +7,115 @@
...
+    hPtYGenD.SetDirectory(0)
+    hPtGenD = hPtYGenD.ProjectionX('hPtGenD')
+elif suffixCF != '':
+    if Dspecie == 'Ds':
+        dirPtShape = infileGenPtShape.Get(f'PWG3_D2H_CFtaskDstoKKpi_CommonFramework_Phi{suffixCF}')
+        contPtShape = dirPtShape.Get(f'CFHFccontainer0_3ProngDstoKKpi_CommonFramework_Phi{suffixCF}')
+        hPtGenD = contPtShape.Project(0, 0)
+        hPtGenD.SetName('hPtGenD')
+    else:
+        print(f'ERROR: D specie {Bspecie} not implemented for CF outputs! Exit')
```
Comment (fcatalan92) [inline systematics/genptshape/ComputePtGenShapeWeights.py:67]: `Bspecie` -> `Dspecie` right?

File: systematics/genptshape/ComputePtGenShapeWeights.py [hunk 2]
```diff
@@ -7,90 +7,115 @@
...
-    else:
-        sTAMU, _ = ReadTAMU('../../models/tamu/D_TAMU_RAA_5TeV_010.txt')
-        sPHSD, _ = ReadPHSD('../../models/phsd/D_PHSD_RAA_5TeV_010.txt')
-        sCatania, _ = ReadCatania('../../models/catania/D_Catania_RAA_5TeV_010.dat')
-    sGossiaux, _ = ReadMCatsHQ('../../models/mcatshq/D_Gossiaux_RAA_5TeV_010.txt')
+isPbPb = False
+sFONLLD, _ = ReadFONLL(shapesD['fonll']['file'], True)
+if Bspecie:
+    sFONLLB, _ = ReadFONLL(shapesB['fonll']['file'], True)
+for shape in shapesD:
```
Comment (fcatalan92) [inline systematics/genptshape/ComputePtGenShapeWeights.py:102]: This loop looks unnecessary if you keep the different if statements

File: systematics/genptshape/ComputePtGenShapeWeights.py [hunk 3]
```diff
@@ -115,38 +140,57 @@
...
+
+            if ptCent < ptMaxGoss:
+                hPtFONLLtimesGossiauxD[-1].SetBinContent(iPt, sFONLLD[pred](ptCent) * sGossiaux['yCent'](ptCent))
+            else:
+                hPtFONLLtimesGossiauxD[-1].SetBinContent(iPt, sFONLLD[pred](ptCent) * sGossiaux['yCent'](ptMaxGoss))
+
+            if ptCent < ptMaxCatania:
+                hPtFONLLtimesCataniaD[-1].SetBinContent(iPt, sFONLLD[pred](ptCent) * sCatania['yCent'](ptCent))
+            else:
+                hPtFONLLtimesCataniaD[-1].SetBinContent(iPt, sFONLLD[pred](ptCent) * sCatania['yCent'](ptMaxCatania))
```
Comment (fcatalan92) [inline systematics/genptshape/ComputePtGenShapeWeights.py:162]: The various `ptMax` are not always defined, they are only if the corresponding model is used

**Review by fcatalan92 — CHANGES_REQUESTED** [review]
Upload also the GenPtShape_LHC20g2a.root and GenPtShape_LHC20g2b.root files please

File: systematics/genptshape/ComputePtGenShapeWeights.py (hunk 2 above)
Comment (fgrosa) [reply systematics/genptshape/ComputePtGenShapeWeights.py:102]: Done

File: systematics/genptshape/ComputePtGenShapeWeights.py (hunk 3 above)
Comment (fgrosa) [reply systematics/genptshape/ComputePtGenShapeWeights.py]: Done

File: systematics/genptshape/ComputePtGenShapeWeights.py (hunk 1 above)
Comment (fgrosa) [reply systematics/genptshape/ComputePtGenShapeWeights.py]: Done

**Comment by fcatalan92** [comment]
Upload also the GenPtShape_LHC20g2a.root and GenPtShape_LHC20g2b.root files please

**Comment by fgrosa** [comment]
> Upload also the GenPtShape_LHC20g2a.root and GenPtShape_LHC20g2b.root files please

The reason why I didn't uploaded them is that they are quite large (~100MB each) and in particular the one for central events exceeds the 100 MB limit of github. In any case, I don't think it is necessary, since we do not store all the `AnalysisResults.root` in the repository

**Comment by fcatalan92** [comment]
> The reason why I didn't uploaded them is that they are quite large (~100MB each) and in particular the one for central events exceeds the 100 MB limit of github. In any case, I don't think it is necessary, since we do not store all the `AnalysisResults.root` in the repository

Ok, we uploaded the ones for the preliminary. It is not necessary although

**Review by fcatalan92 — APPROVED** [review]

**Pull request merged**